### ISSUE TITLE
Avoid cloning in workers

### ISSUE THREAD

**Issue by mllg** [mlr-org/mlr3#180]

**Comment by mllg** [comment]
Instead of cloning, manual filtering is also possible. Carefully benchmark if this is worth the trouble:

```r
  # subset task
  task = data$task
  rows_before = task$row_roles$use
  on.exit({ task$row_roles$use = rows_before })
  task$row_roles$use = intersect(rows_before, e$train_set)
  task$hash = NA_character_
```

**Comment by mllg** [comment]
Solved via 5fcfcd4e795641650fd6ef3bfc93c4afb34485eb.
